lang/funcs/simple/getenv_func.go

```diff
@@ -0,0 +1,43 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// a function to look up variables in the mgmt server instance environment
+
+package simple // TODO: should this be in its own individual package?
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	Register("getenv", &types.FuncValue{
```

Comment:
I think this also needs a `hasenv` function that tells us if an env exists. To differentiate between empty string and non-existant. See: https://golang.org/pkg/os/#LookupEnv

Whether we then use the "default" approach as implemented, or "return empty string" instead, needs discussion.

Maybe both?

Reply:
I have been using an empty string in environment variables, mostly to prevent it from going to the default option, so that might be an explicit feature needed.